README.md

```diff
@@ -1,29 +1,131 @@
+- Every NFT will have a combination of a **`uint256 tokenId`** variable and contract address that must be globally unique.
+- All the fun stuff about NFTs - the rarity, description, image, etc - is part of its **metadata**. In this tutorial, the NFT's **`string tokenURI`** will point to where the metadata is stored on [IPFS](https://ipfs.io/).
+
+## Create the NFT smart contract
+
+Creating a NFT smart contract is short and sweet! [OpenZeppelin](https://www.openzeppelin.com/contracts) contracts make it super easy. First, we'll want change into our `packages/truffle` directory and add the OpenZeppelin dependency:
+
 ```bash
-$ yarn install
+cd packages/truffle
+yarn add @openzeppelin/contracts
 ```
+Then, create a `BoredPetsNFT.sol` file under `packages/truffle/contracts`. :
 
-## Getting Started
+```
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.22 <0.9.0;
 
-- Do this
-- Do that
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
 
-## Contract
+contract BoredPetsNFT is ERC721URIStorage {
+  using Counters for Counters.Counter;
+  Counters.Counter private _tokenIds;
 
-...
+  constructor() ERC721("Bored Pets Yacht Club", "BPYC") {}
+
+  function mint(string memory _tokenURI) public {
+    _tokenIds.increment();
+    uint256 newTokenId = _tokenIds.current();
+    _safeMint(msg.sender, newTokenId);
+    _setTokenURI(newTokenId, _tokenURI);
+  }
+}
+```
+Let's take a look at the imports:
+
+- `@openzeppelin/contracts/token/ERC721/ERC721.sol`
+
+To be a valid NFT, `BoredPetsNFT` implements the ERC-721 standard by inheriting the implementation of `ERC721URIStorage.sol`
+
+- `@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol`
+
+`ERC721URIStorage.sol` is an extension of `ERC721.sol`, which stores the `tokenURI` in `storage` (aka on-chain). If we did not use this extension, the `tokenURI` is derived by concatenating a `baseURI` and `tokenId`. However, we want to store the metadata on IPFS, which identifies content via a hash. Therefore, we cannot generate the `tokenURI` by using the `tokenId`. Instead, we must store the `tokenURI` on-chain to get that data.
+
+To use this, we simply need to code `BoredPetsNFT is ERC721URISTorage`.
+
+- `@openzeppelin/contracts/utils/Counters.sol`
 
-## Front End
+In order to guarantee each `tokenId` is unique, we will increment it by one each time a NFT is minted. We use `Counters.Counter private _tokenIds` to increment and track the value of `tokenId`.
+
+Now, let's go over the methods:
+
+- `constructor() ERC721("Bored Pets Yacht Club", "BPYC")`
+
+The constructor is called when we deploy the smart contract. In this case, `constructor()` indicates our our NFT contract does not take in any parameters. Then, we set "Bored Pets Yacht Club" as the name and "BPYC" as the symbol for our NFTs.
```

Comment:
**Consider:**

The constructor is a function that can be executed upon contract creation. In this case, `constuctor()` indicates that this function is empty, and does not run any contract initialization code. We then set our token collection name to "Bored Pets Yacht Club" and its symbol (abbreviated name) to "BPYC".